pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -447,6 +457,19 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				replicas[i].Name)
 			return &status, nil
 		}
+		// If we have a Pod that has been created but is not available we can not make progress.
+		// We must ensure that all for each Pod, when we create it, all of its predecessors, with respect to its
+		// ordinal, are Available.
+		// Since available is superset of Ready, once we have this featuregate enabled by default, we can remove the
```

Comment:
personally I would prefer TODO or some similar label here :-) but don't know what are the conventions for this..

Reply:
Done. PTAL

Reply:
+1